File: CSharp/SubscriptionApi/SubscriptionApi/Queries/UserQuery.cs
using System.Collections.Generic;
using System.Linq;
using SubscriptionApi.Authorization;

namespace SubscriptionApi.Queries
{
  public class UserQuery : AssetQuery
  {
    private readonly string _name;
    private AssetQueryResolution _authorizationStructure;

    public UserQuery(string name, AssetQueryResolution authorizationStructure)
    {
      _name = name;
      _authorizationStructure = authorizationStructure;
    }

    public void ResolveInto(List<string> requestedAssetNames, QueryResolutionEvents resolutionEvents)
    {
      var assetsFromQuery = _authorizationStructure.RetrieveAssetsByUserName(_name);
      if (!assetsFromQuery.Any())
      {
        resolutionEvents.NoResolutionResultsFor(_name);
      }
      else
      {
        requestedAssetNames.AddRange(assetsFromQuery);
      }
    }
  }
}

File: Java/factories/_01_encapsulation_of_global_context/src/main/java/queries/UserQuery.java
package queries;

import authorization.AssetQueryResolution;

import java.util.List;

public class UserQuery implements AssetQuery {
    private final String name;
    private AssetQueryResolution authorizationStructure;

    public UserQuery(String name, AssetQueryResolution authorizationStructure) {
        this.name = name;
        this.authorizationStructure = authorizationStructure;
    }

    public void resolveInto(List<String> requestedAssetNames, QueryResolutionEvents resolutionEvents) {
        List<String> assetsFromQuery = authorizationStructure.retrieveAssetsByUserName(name);
        if (assetsFromQuery.isEmpty()) {
            resolutionEvents.noResolutionResultsFor(name);
        } else {
            requestedAssetNames.addAll(assetsFromQuery);
        }
    }
}